test/utils/SearchInput-test.ts

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2023 Boluwatife Omosowon <boluomosowon@gmail.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { mocked } from "jest-mock";
+
+import { parsePermalink } from "../../src/utils/permalinks/Permalinks";
+import { transformSearchTerm } from "../../src/utils/SearchInput";
+
+jest.mock("../../src/utils/permalinks/Permalinks");
+
+const searchTerm = "search term";
+const roomLink = "https://matrix.to/#/#element-dev:matrix.org";
+const userLink = "https://matrix.to/#/@element-dev:matrix.org";
+
+const parserLink = {
+    roomLink: "#element-dev:matrix.org",
+    userLink: "@element-dev:matrix.org",
+    searchTerm: null,
+};
+
+describe("transforming search term", () => {
+    it("should return the roomIdOrAlias if the search term was a room permalink", () => {
+        mocked(parsePermalink).mockReturnValue({
+            primaryEntityId: parserLink.roomLink,
+            roomIdOrAlias: parserLink.roomLink,
+            eventId: "",
+            userId: "",
+            viaServers: [],
+            sigil: "",
+        });
+        expect(transformSearchTerm(roomLink)).toBe(parserLink.roomLink);
+    });
+
+    it("should return the userId if the search term was a userId permalink", () => {
+        mocked(parsePermalink).mockReturnValue({
+            primaryEntityId: parserLink.userLink,
+            roomIdOrAlias: "",
+            eventId: "",
+            userId: parserLink.userLink,
+            viaServers: [],
+            sigil: "",
+        });
+        expect(transformSearchTerm(userLink)).toBe(parserLink.userLink);
+    });
```

Comment:
I've just been familiarising myself with the timeline of work on this PR and see these tests were added towards the end.

Just a question on these two tests. Here we're mocking what we get back from the call to `parsePermalink`. However whilst the tests say `should return roomIdOrAlias` and `should return userId` it looks to me from the code like we're returning `primaryEntityId` in both cases (which matches both of these fields in both tests).

To me it seems that we could replace these tests with a single `returns the primaryEntityId if the search term was a permalink` and still have the same coverage. What do you think?

Reply:
A single test seems okay. I only made it this way for more descriptive tests.  I will work on it shortly.

Reply:
When you do get around to working on this @bolu-tife , we're also trying to make sure we're not adding new TS issues in PRs. If you could address the (single) TS failure in this PR that would be much appreciated.

Reply:
okay, I would look into that as well